lib/Notification/Notifier.php

```diff
@@ -91,10 +97,32 @@ public function prepare(INotification $notification, string $languageCode): INot
 		switch ($notification->getSubject()) {
 			case self::SUBJECT:
 				$parameters = $notification->getSubjectParameters();
+				$message = $parameters[0];
 				$notification->setParsedSubject($l->t('Nextcloud announcement'))
-					->setParsedMessage($parameters[0])
 					->setIcon($this->url->getAbsoluteURL($this->url->imagePath($this->appName, 'app-dark.svg')));
 
+				$action = $notification->createAction();
+				$action->setParsedLabel($l->t('Read more'))
+					->setLink($notification->getLink(), IAction::TYPE_WEB)
+					->setPrimary(true);
+				$notification->addParsedAction($action);
+
+				$isAdmin = $this->groupManager->isAdmin($notification->getUser());
+				if ($isAdmin) {
+					$groups = $this->config->getAppValue($this->appName, 'notification_groups', '');
+					if ($groups === '') {
+						$action = $notification->createAction();
+						$action->setParsedLabel($l->t('Disable announcements'))
+							->setLink($this->url->linkToOCSRouteAbsolute('provisioning_api.AppsController.disable', ['app' => 'nextcloud_announcements']), IAction::TYPE_DELETE)
+							->setPrimary(false);
+						$notification->addParsedAction($action);
+
+						$message .= "\n\n" . $l->t('(These announcements are only shown to administrators)');
```

Comment:
```suggestion
						$message .= "\n\n" . $l->t('(These announcements are only shown to administrators.)');
```

Very small nitpick. ;D

Reply:
I had it in place before, found it looked weird and removed it, damn :D

Reply:
Yeah, it does look a bit strange and unfinished cause the sentences of the announcement will all have punctuation. As said, just a tiny detail. :)